// @hubspot/api-client/lib/src/discovery/crm/objects/ObjectsDiscovery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectsDiscovery = void 0;
const configuration_1 = require("../../../../codegen/crm/objects/configuration");
const index_1 = require("../../../../codegen/crm/objects/index");
const rxjsStub_1 = require("../../../../codegen/crm/objects/rxjsStub");
const ApiClientConfigurator_1 = require("../../../configuration/ApiClientConfigurator");
class ObjectsDiscovery {
    constructor(config) {
        const configuration = configuration_1.createConfiguration(ApiClientConfigurator_1.ApiClientConfigurator.getParams(config, rxjsStub_1.Observable, rxjsStub_1.Observable));
        this.associationsApi = new index_1.AssociationsApi(configuration);
        this.basicApi = new index_1.BasicApi(configuration);
        this.batchApi = new index_1.BatchApi(configuration);
        this.gdprApi = new index_1.GDPRApi(configuration);
        this.searchApi = new index_1.SearchApi(configuration);
    }
}
exports.ObjectsDiscovery = ObjectsDiscovery;
