// @hubspot/api-client/lib/codegen/webhooks/types/PromiseAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromiseSubscriptionsApi = exports.PromiseSettingsApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class PromiseSettingsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableSettingsApi(configuration, requestFactory, responseProcessor);
    }
    clear(appId, appId2, _options) {
        const result = this.api.clear(appId, appId2, _options);
        return result.toPromise();
    }
    configure(appId, appId2, settingsChangeRequest, _options) {
        const result = this.api.configure(appId, appId2, settingsChangeRequest, _options);
        return result.toPromise();
    }
    getAll(appId, appId2, _options) {
        const result = this.api.getAll(appId, appId2, _options);
        return result.toPromise();
    }
}
exports.PromiseSettingsApi = PromiseSettingsApi;
const ObservableAPI_2 = require("./ObservableAPI");
class PromiseSubscriptionsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableSubscriptionsApi(configuration, requestFactory, responseProcessor);
    }
    archive(subscriptionId, appId, appId2, _options) {
        const result = this.api.archive(subscriptionId, appId, appId2, _options);
        return result.toPromise();
    }
    create(appId, appId2, subscriptionCreateRequest, _options) {
        const result = this.api.create(appId, appId2, subscriptionCreateRequest, _options);
        return result.toPromise();
    }
    getAll(appId, appId2, _options) {
        const result = this.api.getAll(appId, appId2, _options);
        return result.toPromise();
    }
    getById(subscriptionId, appId, appId2, _options) {
        const result = this.api.getById(subscriptionId, appId, appId2, _options);
        return result.toPromise();
    }
    update(subscriptionId, appId, appId2, subscriptionPatchRequest, _options) {
        const result = this.api.update(subscriptionId, appId, appId2, subscriptionPatchRequest, _options);
        return result.toPromise();
    }
    updateBatch(appId, appId2, batchInputSubscriptionBatchUpdateRequest, _options) {
        const result = this.api.updateBatch(appId, appId2, batchInputSubscriptionBatchUpdateRequest, _options);
        return result.toPromise();
    }
}
exports.PromiseSubscriptionsApi = PromiseSubscriptionsApi;
